homeassistant/components/deconz/gateway.py

```diff
@@ -66,24 +72,24 @@ def __init__(self, hass, config_entry) -> None:
             sensor.RESOURCE_TYPE: self.signal_new_sensor,
         }
 
-        self.deconz_ids = {}
-        self.entities = {}
-        self.events = []
+        self.deconz_ids: dict[str, str] = {}
+        self.entities: dict[str, set[str]] = {}
+        self.events: list[DeconzAlarmEvent | DeconzEvent] = []
 
     @property
     def bridgeid(self) -> str:
         """Return the unique identifier of the gateway."""
-        return self.config_entry.unique_id
+        return cast(str, self.config_entry.unique_id)
```

Comment:
It might be better to `assert self.config_entry.unique_id is not None` instead of the cast. That way the type-checker and the run-time environment are operating under the same assumptions. And if it turns out to actually be `None`, you'll find the cause of any issue before it manifests itself in a consumer of the `bridgeid` property.

Reply:
I was asked in previous PRs related to typing Deconz not to use assert as it could lead to unintended effects

Reply:
Interesting. I know asserts can cause warnings to be logged and annoy users if they ever occur, but thought that would be better than trying to track down bugs that appear later on when the bad return value is finally used.

I'll defer to the previous advice.